mlir/utils/jenkins/clang-format.ignore

```diff
@@ -0,0 +1 @@
+# Patterns for clang-format to ignore.
```

Comment:
Now on a second look, it probably makes more sense to move scripts and ignore files under a dedicated folder, for example, `mlir/utils/jenkins/static-checks/`